rotkehlchen/exchanges/iconomi.py

```diff
@@ -0,0 +1,260 @@
+        tx_amount = raw_trade['source_amount']
+        native_amount = raw_trade['target_amount']
+        native_asset = Asset(raw_trade['target_ticker'])
+
+    pair = TradePair(f'{tx_asset.identifier}_{native_asset.identifier}')
+    amount = tx_amount
+    rate = Price(native_amount / tx_amount)
+    fee_amount = raw_trade['fee_amount']
+    fee_asset = Asset(raw_trade['fee_ticker'])
+
+    return Trade(
+        timestamp=timestamp,
+        location=Location.ICONOMI,
+        pair=pair,
+        trade_type=trade_type,
+        amount=amount,
+        rate=rate,
+        fee=fee_amount,
+        fee_currency=fee_asset,
+        link=str(raw_trade['transactionId']),
+    )
+
+
+class Iconomi(ExchangeInterface):
+    def __init__(
+            self,
+            api_key: ApiKey,
+            secret: ApiSecret,
+            database: 'DBHandler',
+            msg_aggregator: MessagesAggregator,
+    ):
+        super().__init__('iconomi', api_key, secret, database)
+        self.uri = 'https://api.iconomi.com'
+        self.session.headers.update({'ICN-API-KEY': api_key})
+        self.msg_aggregator = msg_aggregator
+
+    def _generate_signature(self, request_type: str, request_path: str, timestamp: str) -> None:
+        signed_data = ''.join([timestamp, request_type.upper(), request_path, '']).encode()
+        signature = hmac.new(
+            self.secret,
+            signed_data,
+            hashlib.sha512,
+        )
+        self.session.headers.update({
+            'ICN-SIGN': base64.b64encode(signature.digest()).decode(),
+        })
+
+    def _api_query(
+            self,
+            verb: str,
+            path: str,
+            options: Optional[Dict] = None,
+    ) -> Dict:
+        """
+        Queries ICONOMI with the given verb for the given path and options
+        """
+        assert verb in ('get', 'post', 'push'), (
+            'Given verb {} is not a valid HTTP verb'.format(verb)
+        )
+
+        request_path_no_args = '/v1/' + path
+
+        data = ''
+        if not options:
+            request_path = request_path_no_args
+        else:
+            request_path = request_path_no_args + '?' + urlencode(options)
+
+        timestamp = str(int(time.time() * 1000))
+        request_url = self.uri + request_path
+
+        self._generate_signature(
+            request_type=verb.upper(),
+            request_path=request_path_no_args,
+            timestamp=timestamp,
+        )
+
+        self.session.headers.update({
+            'ICN-TIMESTAMP': timestamp,
+        })
+        if data != '':
+            self.session.headers.update({
+                'Content-Type': 'application/json',
+                'Content-Length': str(len(data)),
+            })
+
+        log.debug('ICONOMI API Query', verb=verb, request_url=request_url)
+
+        try:
+            response = getattr(self.session, verb)(request_url, data=data, timeout=30)
+        except ReadTimeout as exc:
```

Comment:
Why only ReadTimeout? I would suggest catching all possible request exceptions with `requests.exceptions.RequestException`, logging an error and as you already do raise a RemoteError.

Essentially same thing we do in other exchanges. Example:

https://github.com/rotki/rotki/blob/4500381d5c527d255256f1a32d8261722ce028c2/rotkehlchen/exchanges/bittrex.py#L288-L295